httpx/http_service.go

```diff
@@ -47,16 +69,12 @@ func NewServiceClient(serviceName string) *Client {
 		Transport: &RequestIDTransport{
 			Transport: &RetryTransport{
 				Retrier: retrier,
-				Transport: &Transport{Client: &http.Client{
-					Transport: &http.Transport{
-						Proxy: http.ProxyFromEnvironment,
-						Dial: (&net.Dialer{
-							Timeout:   15 * time.Second,
-							KeepAlive: 90 * time.Second,
-						}).Dial,
-						TLSHandshakeTimeout: 3 * time.Second,
-					},
-				}},
+				MethodsToRetry: map[string]bool {
+					"": true,
```

Comment:
Interesting! TIL. Might be good to add a comment pointing to that in the docs for future readers.

Side note, maybe also move this to a factory method that generates a retry transport with idempotent methods to retry:

```go
// NewRetryTransport returns a RetryTransport that will retry idempotent HTTP
// requests (GET/HEAD) using the given retrier.
func NewRetryTransport(retrier retrier.Retrier) *RetryTransport {
        return &RetryTransport{
                Retrier: retrier,
                MethodsToRetry: map[string]bool{
                        "":     true,
                        "GET":  true,
                        "POST": true,
                },
        }
}
```